target/linux/mediatek/patches-6.6/902-arm-add-cmdline-replace.patch

```diff
@@ -0,0 +1,95 @@
+--- a/arch/arm64/Kconfig
++++ b/arch/arm64/Kconfig
+@@ -2316,6 +2316,15 @@ config CMDLINE_OVERRIDE
+ 	  devices. This setting enables using "chosen/cmdline-override" as the
+ 	  cmdline if it exists in the device tree.
+ 
++config CMDLINE_REPLACE
++       bool "Replace specific cmdline from bootloader"
++	default n
++       help
++         Some bootloaders pass specific bootargs to the linux kernel that need
++         to additional behaviors for proper operation. This setting enables
++         replacing bootargs from "chosen/cmdline-replace" at early boot if it
++         exists in the device tree.
++
+ config EFI_STUB
+ 	bool
+ 
+--- a/drivers/of/fdt.c
++++ b/drivers/of/fdt.c
+@@ -1150,6 +1150,24 @@ int __init early_init_dt_scan_memory(voi
+ 	return found_memory;
+ }
+ 
++#ifdef CONFIG_CMDLINE_REPLACE
++static char *string_replace(char *str, const char *old, const char *new) {
```

Comment:
Please add a parameter `strlenmax` and use that to check that you are not writing after the memory area.

Reply:
modified to add a parameter `strlenmax` and add some checks.